e2e/framework/manager/rook_test_infra.go

```diff
@@ -319,14 +327,17 @@ func createK8sRookClient(k8sHelper *utils.K8sHelper, tag string) (err error) {
 	return nil
 }
 
-func createk8sRookCluster(k8sHelper *utils.K8sHelper) error {
+func createk8sRookCluster(k8sHelper *utils.K8sHelper, tag string) error {
 	raw, err := ioutil.ReadFile(podSpecPath + "/" + rookClusterFileName)
 
 	if err != nil {
 		return err
 	}
-
-	rookCluster := string(raw)
+	if !bytes.Contains(raw, []byte(defaultTagName)) {
+		return fmt.Errorf(" %s tag to be replaced with  %s couldn't be found in rook-cluster.yaml", defaultTagName, tag)
+	}
+	rawUpdated := bytes.Replace(raw, []byte(defaultTagName), []byte(tag), 1)
```

Comment:
we are using bytes[] since the ioutils.ReadFile returns a byte[]. We would like to keep it as a byte[] so we know where are checking or replacing contents of a file.